articles/how-to-consume-data-from-an-API-using-react/index.md

```diff
@@ -0,0 +1,362 @@
+
+This method helps in updating the search query state when the value of the input changes, The value of the search state is updated using the `setSearch` method.
+
+```JavaScript
+const onInputChange = e => {
+   setSearch(e.target.value);
+ }
+```
+
+### Step 6: Returning the component
+
+After we fetch data, we should display it to the user. In the `app.js` file, we should include an input field and a button for search. The page also has an image, h4, and paragraph tags. Here is the code for our app's layout.
+
+```javascript
+return(
+   <div>
+     <input type="text" value={search} onChange={onInputChange}/>
+     <button type="submit" onClick={getMovie}>Search</button>
+     
+     <img src={movie.Poster} alt=""/>
+     <h4>Title: {movie.Title}</h4>
+     <p>Year: {movie.Year}</p>
+     <p>Writer: {movie.Writer}</p>
+   </div>
+);
+```
+
+As shown above, the `getMovie` method is called whenever the button is clicked. 
+
+The `onInputChange` function is called when the value of the input changes.
+
+Since we are returning a JSON object rather than an array, there is no need to loop through it. We use `movie.Title`,`movie.Year`, and `movie.Writer` to extract data and pass it to the children components.
+
+Here is the code for the `app.js` component
+
+```javascript
+import './App.css';
+import React, {useEffect, useState} from 'react';
+
+function App() {
+    const [movie, setMovie] = useState([]);
+    const [search, setSearch] = useState('');
+    const API_KEY = "cebd9b53";
+    const url = `http://www.omdbapi.com/?t=${search}&apikey=${API_KEY}`;
+
+    const getMovie = async()=>{
+      const response = await fetch(url);
+      const data = await response.json()
+      setMovie(data);
+    }
+
+    const onInputChange = e =>{
+      setSearch(e.target.value);
+    }
+
+    useEffect(()=> {
+      getMovie();
+    }, []);
+
+    return(
+      <div>
+        <input type="text" value={search} onChange={onInputChange}/>
+        <button type="submit" onClick={getMovie}>Search</button>
+            <br></br>
+            if(movie==null){
+              <p>Movie Not Found</p>
+            }else{
+            <img src={movie.Poster} alt=""/>
+            <h4>Title: {movie.Title}</h4>
+            <p>Year: {movie.Year}</p>
+            <p>Writer: {movie.Writer}</p>
+            }
+      </div>
+    );
+
+}
+
+export default App;
+
+```
+### Step 7: Handling errors
+Errors are a common occurence when dealing with an API. For this tutorial, we need to notify the user in case a movie is not found in the database. One key variable we can use to track the data is `Response`. The API returns the `Response` with a value of `True` when a movie is found and `False` if its unavailable. We, therefore, check the state of this variable.
+
+To do this, we need a new method and an if else statement. create a method named checkResponse in the app.js file. Add the following code.
+
+```JSX
+function checkResponse(data){
+  if(data.Response==="True"){
```

Comment:
What do you do in case there is no response from the API due to network issues? 

You should discuss error handling with `try/catch` blocks in case there is a network error or server error.